include/dp_flow.h

```diff
@@ -69,6 +69,9 @@ enum dp_flow_tcp_state {
 };
 
 struct flow_key {
+	// my proposed refactor would be much simpler if there could be l3_type for *both* l3_src and l3_dst
+	// (as part of a common packed struct dp_ip_addr_key)
+	// I realize that having DST and SRC types differ makes no sense, so it would waste one byte, but all handling functions would be cleaner
```

Comment:
This is just a question for the refactor I have prepared. Since more places use the IPv4/IPv6 in a hash key, I would have a common structure (packed) to use along with handler functions that take care of the whole "make sure you zero the rest and set type properly". 

No I would like to also use these safer wrappers for `flow_key` but it only has one type and two IPs, thus I would propose to sacrifice one byte to simply have the two common structures used.

Reply:
Yes, it is ok to sacrifice 1 byte for ease of implementation and consistency.